src/components/application_manager/test/commands/hmi/navi_audio_start_stream_request_test.cc

```diff
@@ -0,0 +1,357 @@
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(MockAppPtr()));
+  EXPECT_CALL(mock_app_manager_, HMILevelAllowsStreaming(_, _)).Times(0);
+  EXPECT_CALL(*mock_app_, set_audio_streaming_approved(_)).Times(0);
+
+  Event event(hmi_apis::FunctionID::Navigation_StartAudioStream);
+  event.set_smart_object(*msg_);
+
+  command->on_event(event);
+}
+
+TEST_F(AudioStartStreamRequestTest, OnTimeOut_Terminated) {
+  (*msg_)[strings::params][strings::correlation_id] = kCorrelationID;
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+  (*msg_)[strings::params][strings::connection_key] = 3u;
+
+  SharedPtr<AudioStartStreamRequest> command(
+      CreateCommand<AudioStartStreamRequest>(msg_));
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(MockAppPtr()));
+
+  EXPECT_CALL(mock_app_manager_, TerminateRequest(3u, kCorrelationID));
+
+  command->onTimeOut();
+}
+
+TEST_F(AudioStartStreamRequestTest, RetryStartSession_Approved_SUCCESS) {
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  SharedPtr<AudioStartStreamRequest> command(
+      CreateCommand<AudioStartStreamRequest>(msg_));
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, set_audio_stream_retry_number(0));
+
+  command->RetryStartSession();
+}
+
+TEST_F(AudioStartStreamRequestTest, RetryStartSession_NotApproved_SUCCESS) {
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  SharedPtr<AudioStartStreamRequest> command(
+      CreateCommand<AudioStartStreamRequest>(msg_));
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(false));
+  EXPECT_CALL(*mock_app_, audio_stream_retry_number()).WillOnce(Return(0));
+
+  MockMessageHelper& message_helper = *MockMessageHelper::message_helper_mock();
+  EXPECT_CALL(message_helper, SendAudioStartStream(kAppID, _));
+  EXPECT_CALL(*mock_app_, set_audio_stream_retry_number(1));
+
+  command->RetryStartSession();
+}
+
+TEST_F(AudioStartStreamRequestTest, RetryStartSession_NotApproved_Canceled) {
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  SharedPtr<AudioStartStreamRequest> command(
+      CreateCommand<AudioStartStreamRequest>(msg_));
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(false));
+  EXPECT_CALL(*mock_app_, audio_stream_retry_number()).WillOnce(Return(2));
```

Comment:
@indlu , `2` is magic number. for example create a variable `retry_number = 2` and instead of `2`